modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/render/DDMFormFieldFreeMarkerRenderer.java

```diff
@@ -750,6 +736,34 @@ private Locale _getPreferredLocale(
 		return iterator.next();
 	}
 
+	private TemplateResource _updateTemplateResource(
+			String fieldNamespace, String templateName,
+			TemplateResource templateResource)
```

Comment:
You're passing templateResource but where are you using it?

Also can this method be inline?

Reply:
Inline -> changed.

templateResource has the default template (text.ftl) so can be returned if the conditions don't work.